pkg/cache/snapshot.go

```diff
@@ -25,10 +25,12 @@ import (
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 	"sigs.k8s.io/kueue/pkg/features"
 	"sigs.k8s.io/kueue/pkg/hierarchy"
+	"sigs.k8s.io/kueue/pkg/queue"
```

Comment:
Don't add this dependency.

Instead place AfsResourceWeights in some dedicated package like `pkg/util/afs`, similarly as we do for TAS. Then, you can name it without "Afs" in name.

Reply:
There is really no conceptual dependency between dependencies, and for the ease of development there used to be cases the dependency would go either way. So, I prefer to just keep them independent.